Algorithms.Tests/Numeric/EulerMethodTest.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using Algorithms.Numeric;
+using NUnit.Framework;
+using System.Collections.Generic; 
+
+namespace Algorithms.Tests.Numeric
+{
+    public static class EulerMethodTest
+    {
+        [Test]
+        public static void TestLinearEquation()
+        {
+            Func<double, double, double> exampleEquation1 = (x, y) => x;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 4, 0.1, 0, exampleEquation1);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 7.800000000000003);
+        }
+        
+        [Test]
+        public static void TestExampleWikipedia()
+        {
+            // example from https://en.wikipedia.org/wiki/Euler_method
+            Func<double, double, double> exampleEquation2 = (x, y) => y;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 4, 0.1, 1, exampleEquation2);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 45.25925556817596);

```

Comment:
From Wikipedia, error for this example is 9.34, which is huge as a relative error. Please use small steps to get correct answers.